crates/ark/src/modules/positron/graphics.R

```diff
@@ -1,113 +1,69 @@
-    }
-}
-
-renderWithPlotDevice <- function(filepath, format, width, height, res, type) {
-    # width and height are in inches and use 72 DPI to create the requested size in pixels
-    dpi <- 72
+# A persistent list mapping plot `id`s to their display list recording.
+# Used for replaying recordings under a new device or new width/height/resolution.
+RECORDINGS <- list()
 
-    # Create a new graphics device.
-    switch(
-        format,
-        "png" = grDevices::png(
-            filename = filepath,
-            width = width,
-            height = height,
-            res = res,
-            type = type
-        ),
-        "svg" = grDevices::svg(
-            filename = filepath,
-            width = (width / dpi),
-            height = (height / dpi),
-        ),
-        "pdf" = grDevices::pdf(
-            file = filepath,
-            width = (width / dpi),
-            height = (height / dpi)
-        ),
-        "jpeg" = grDevices::jpeg(
-            filename = filepath,
-            width = width,
-            height = height,
-            res = res,
-            type = type
-        ),
-        "tiff" = grDevices::tiff(
-            filename = filepath,
-            width = width,
-            height = height,
-            type = type
-        ),
-        stop("Internal error: Unknown plot `format`.")
-    )
+# Retrieves a recording by its `id`
+#
+# Returns `NULL` if no recording exists
+getRecording <- function(id) {
+    RECORDINGS[[id]]
 }
 
-#' @export
-.ps.graphics.defaultResolution <- if (Sys.info()[["sysname"]] == "Darwin") {
-    96L
-} else {
-    72L
+addRecording <- function(id, recording) {
+    RECORDINGS[[id]] <<- recording
 }
 
-#' @export
-.ps.graphics.plotSnapshotRoot <- function(...) {
-    file.path(tempdir(), "positron-snapshots", ...)
-}
+# TODO: Use this when we get notified that we can remove a recording
+# removeRecording <- function(id) {
+#     RECORDINGS[[id]] <<- NULL
+# }
 
-#' @export
-.ps.graphics.plotSnapshotPath <- function(id) {
-    root <- .ps.graphics.plotSnapshotRoot(id)
+plotRecordingRoot <- function() {
+    root <- file.path(tempdir(), "positron-plot-recordings")
     ensure_directory(root)
-    file.path(root, "snapshot.rds")
+    root
 }
 
-#' @export
-.ps.graphics.plotOutputPath <- function(id) {
-    root <- .ps.graphics.plotSnapshotRoot(id)
-    ensure_directory(root)
-    file.path(root, "snapshot.png")
+plotRecordingPath <- function(id) {
+    root <- plotRecordingRoot()
+    file <- paste0("recording-", id, ".png")
+    file.path(root, file)
 }
 
 #' @export
-.ps.graphics.createDevice <- function(name, type, res) {
-    # Get path where plots will be generated.
-    plotsPath <- .ps.graphics.plotSnapshotRoot("current-plot.png")
-    ensure_parent_directory(plotsPath)
+.ps.graphics.createDevice <- function(name, type) {
+    # Get path where non-recorded plots will be generated.
+    root <- plotRecordingRoot()
+    filename <- file.path(root, "current-plot.png")
```

Comment:
I don't fully understand when this gets written to. It is a side effect of the fact that we piggy back on a `png()` device. Possibly it writes to it anytime we change devices.